nextcloud-startup-script.sh

```diff
@@ -398,6 +398,9 @@ then
 fi
 clear
 
+# We need to unset the cached admin-user since we have changed its name
+unset NC_ADMIN_USER
```

Comment:
we should unset the admin_user if after it gets renamed

Reply:
So we would need to unset it in every script `notify_admin_gui` is used?

Reply:
no, we don't. it will be empty by default. This is the only place where it makes sense to unset this variable.

Reply:
Why does it only make sense to add it in the update script? 

Do you mean that until it's unset, it's cached?

Reply:
No, what I mean, is that we cache the name of admin users. And if we change the name of an admin user, we also need to get the new admin username. we could alternatively just put that new name of the admin user into the admin username array.

Reply:
I might be tired, but I don't get it.

1. You say the array does the caching
2. You unset the array in the end of the update script

Now,
1. Isn't the cache expired every time the function runs?
2. Is the cache expired only if you unset the variable?

Reply:
> Isn't the cache expired every time the function runs?

The cache isn't expire every time the function runs.

> Is the cache expired only if you unset the variable?

Yes because the admin user array is global and we don't overwrite it automatically if it is not empty.

Does this now be a bit clearer?

Reply:
Yes, and that's what I thought. Hence my question:

> Do you mean that until it's unset, it's cached?

You said no - hence my confusion. :)

And, in that case, I think it makes sense to expire the cache in all scripts that use this function.

Reply:
If we expire the cache at the beginning of every script that uses this function then caching doesn't make much sense, does it?

Reply:
I thought in the end, not the beginning. But OK, let's merge as is. Users should run the update script frequently anyway - and then the cache is expired.